jooqdb/src/main/java/org/cristalise/storage/jooqdb/clusterStore/JooqHistoryHandler.java

```diff
@@ -191,6 +196,7 @@ public void createTables(DSLContext context) {
         .column(TARGET_STATE_ID,      ID_TYPE        .nullable(false))
         .column(TRANSITION_ID,        ID_TYPE        .nullable(false))
         .column(VIEW_NAME,            NAME_TYPE      .nullable(true))
+        .column(HAS_ATTACHMENT,       SQLDataType.BOOLEAN.nullable(false))
```

Comment:
Should be `.nullable(true)`, because the already existing events will have null in this newly added field (unless you set a default FALSE value when adding the new field with `ALTER TABLE`). This might be the reason for the problems we're experiencing upon startup.

Reply:
Already set default value false for alter table

Reply:
pls write the code that it will work without ALTER TABLE. there will be project who does want to control when this feature is enabled.

You need to make it nullable, and use a system property to disable this feature.

Reply:
@taurus227

Reply:
No need to make it nullable. But it does need a default value.